projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-file-view/components/case-file-view-folder/case-file-view-folder.component.ts

```diff
@@ -0,0 +1,251 @@
+
+  @Input() public categoriesAndDocuments$: Observable<CategoriesAndDocuments>;
+
+  public nestedTreeControl: NestedTreeControl<DocumentTreeNode>;
+  public nestedDataSource: DocumentTreeNode[];
+  public categories: CaseFileViewCategory[] = [];
+  public categoriesAndDocumentsSubscription: Subscription;
+
+  private getChildren = (node: DocumentTreeNode) => of(node.children);
+  public hasNestedChild = (_: number, nodeData: DocumentTreeNode) => nodeData.children;
+
+  constructor() {
+    this.nestedTreeControl = new NestedTreeControl<DocumentTreeNode>(this.getChildren);
+  }
+
+  public ngOnInit(): void {
+    this.categoriesAndDocumentsSubscription = this.categoriesAndDocuments$
+      .subscribe(categoriesAndDocuments => {
+      // Using the mock data for now as we have to display the documents as well for demo purpose
+      const categories = this.loadCategories(); // categoriesAndDocuments.categories;
+      // Generate document tree data from categories
+      const treeData = this.generateTreeData(categories);
+      // Append uncategorised documents
+      if (categoriesAndDocuments.uncategorised_documents && categoriesAndDocuments.uncategorised_documents.length > 0) {
+        const uncategorisedDocuments = this.getUncategorisedDocuments(categoriesAndDocuments.uncategorised_documents);
+        treeData.push(uncategorisedDocuments);
+      }
+      // Initialise cdk tree with generated data
+      this.nestedDataSource = treeData;
+    });
+  }
+
+  public generateTreeData(categories: CaseFileViewCategory[]): DocumentTreeNode[] {
+    return categories.reduce((tree, node) => [
+      ...tree,
+      ...[
+        plainToClass(DocumentTreeNode, {
+          name: node.category_name,
+          type: 'category',
+          children: [...this.generateTreeData(node.sub_categories), ...this.getDocuments(node.documents)],
+        })
+      ],
+    ], []);
+  }
+
+  public getDocuments(documents: CaseFileViewDocument[]): DocumentTreeNode[] {
+    const documentsToReturn: DocumentTreeNode[] = [];
+    documents.forEach(document => {
+      documentsToReturn.push(plainToClass(DocumentTreeNode,
+        { name: document.document_filename, type: 'document' })
+      );
+    });
+
+    return documentsToReturn;
+  }
+
+  public getUncategorisedDocuments(uncategorisedDocuments: CaseFileViewDocument[]): DocumentTreeNode {
+    const documents: DocumentTreeNode[] = [];
+    uncategorisedDocuments.forEach(document => {
+      documents.push(plainToClass(DocumentTreeNode, {name: document.document_filename, type: 'document' }));
+    });
+    return plainToClass(DocumentTreeNode,
+      { name: CaseFileViewFolderComponent.UNCATEGORISED_DOCUMENTS_TITLE, type: 'category', children: documents }
+    );
+  }
+
+  public ngOnDestroy(): void {
+    if (this.categoriesAndDocumentsSubscription) {
+      this.categoriesAndDocumentsSubscription.unsubscribe();
+    }
+  }
+
+  public loadCategories(): CaseFileViewCategory[] {
```

Comment:
Why are we loading test data here when we have an input for the data we need? If test data is needed, it should still be pushed in through the input so that test-data only exists in 1 place (the container component)

Reply:
The actual data returned by the API does not contain any documents. I had to work on a ticket to display folder and document icons and had to use this dummy data for the moment.

Dan will soon be adding documents so that will be returned by the API and this dummy data can be deleted.

I have added a comment at line 38 in this file.

Reply:
Even then, the mock data should live in the service, not the component. The service is responsible for returning mock data if API is not yet available, so there is only one place to make a change later on.